### PR TITLE
Make sure sibling images get a link too

### LINKED ISSUE #12925: If adding two images in the description when creating a new issue, only the first is clickable.
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes https://try.gitea.io/floaf/test/issues/2
  - [ ] No
  - [ ] Not relevant

If adding two images (by pasting them in the text field) in the description when creating a new issue, only the first is clickable.

image
image
Only 1 is clickable, not 2.

### PR DESCRIPTION
**Pull request by zeripath**
Due a problem with the ast.Walker in the our transformer in goldmark
an image with a sibling image will not be transformed to gain a parent
link. This PR fixes this.

Fix #12925

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by noerw** [comment]
Could you add a testcase, I'm struggling to understand what this is for ;)

**Review by lafriks — APPROVED** [review]

**Review by noerw — APPROVED** [review]
thanks :>

**Comment by noerw** [comment]
testcase is missing a `\n` at the end 🙃

File: modules/markup/markdown/markdown_test.go [hunk 1]
```diff
@@ -308,3 +308,13 @@ func TestRender_RenderParagraphs(t *testing.T) {
 	test(t, "A\n\nB\nC\n", 2)
 	test(t, "A\n\n\nB\nC\n", 2)
 }
+
+func TestRenderSiblingImages_Issue12925(t *testing.T) {
+	testcase := `![image1](/image1)
+![image2](/image2)
+`
+	expected := `<p><a href="/image1" rel="nofollow"><img src="/image1" alt="image1"></a><br>
+<a href="/image2" rel="nofollow"><img src="/image2" alt="image2"></a></p>`
```
Comment (zeripath) [inline modules/markup/markdown/markdown_test.go]:
```suggestion
<a href="/image2" rel="nofollow"><img src="/image2" alt="image2"></a></p>
`
```

**Comment by zeripath** [comment]
weird it worked locally in vscode

**Pull request merged**